src/main/java/com/bravo/user/controller/UserController.java

```diff
@@ -45,18 +45,32 @@ public UserReadDto create(final @RequestBody UserSaveDto request, final BindingR
 
   @GetMapping(value = "/retrieve/{id}")
   @ResponseBody
-  public UserReadDto retrieve(@PathVariable String id){
+  public UserReadDto retrieve(@PathVariable String id) {
     return userService.retrieve(id);
   }
 
+  @GetMapping(value = "/retrieve")
+  @ResponseBody
+  public List<UserReadDto> retrieveByName(
+      final @RequestParam String name,
+      final @RequestParam(required = false) Integer page,
+      final @RequestParam(required = false) Integer size,
+      final HttpServletResponse httpResponse
+  ) {
+    // validate name but allow control characters
+    userValidator.validateName(name.replaceAll("[!*%]", ""));
```

Comment:
I like that the validation is first.  The replaceAll... should that be a method that other calls can utilize?    Also, is there a potential NPE here?

Reply:
In practice I think Spring would reject the call if `name` were not passed since is a required parameter.  I'm not sure that we'd ever actually see an NPE here, but a null check would be a good idea.  I've added that along with moving the `replaceAll` method to validator utils.